addons/stock_account/models/stock.py

```diff
@@ -270,32 +264,30 @@ def _get_accounting_data_for_valuation(self):
             raise UserError(_('You don\'t have any stock valuation account defined on your product category. You must define one before processing this operation.'))
         journal_id = accounts_data['stock_journal'].id
         return journal_id, acc_src, acc_dest, acc_valuation
-
+    
     def _prepare_account_move_line(self, qty, cost, credit_account_id, debit_account_id):
         """
         Generate the account.move.line values to post to track the stock valuation difference due to the
         processing of the given quant.
         """
         self.ensure_one()
-
+    
         if self._context.get('force_valuation_amount'):
             valuation_amount = self._context.get('force_valuation_amount')
         else:
-            if self.product_id.cost_method == 'average':
-                valuation_amount = cost if self.location_id.usage == 'supplier' and self.location_dest_id.usage == 'internal' else self.product_id.standard_price
-            else:
-                valuation_amount = cost if self.product_id.cost_method == 'real' else self.product_id.standard_price
+            valuation_amount = cost
+
         # the standard_price of the product may be in another decimal precision, or not compatible with the coinage of
         # the company currency... so we need to use round() before creating the accounting entries.
-        debit_value = self.company_id.currency_id.round(valuation_amount * qty)
-
+        debit_value = self.company_id.currency_id.round(valuation_amount)
+    
         # check that all data is correct
         if self.company_id.currency_id.is_zero(debit_value):
             if self.product_id.cost_method == 'standard':
                 raise UserError(_("The found valuation amount for product %s is zero. Which means there is probably a configuration error. Check the costing method and the standard price") % (self.product_id.name,))
             return []
         credit_value = debit_value
-
+    
         if self.product_id.cost_method == 'average' and self.company_id.anglo_saxon_accounting:
             # in case of a supplier return in anglo saxon mode, for products in average costing method, the stock_input
             # account books the real purchase price, while the stock account books the average price. The difference is
```

Comment:
@sle-odoo Thanks, for you reply,

I am looking closely at this MP because the way this MP is released to master will impact our way of providing services to our customers. It will make easier some task or will lead us  to do other task in a different way, or will cause that we have to develop new approaches to existing problems.

It is good to have an early contact with you as you are in the drawing board, because we can exchange ideas with you and your team.

Regards.